src/api/uploads/PlainUpload.js

```diff
@@ -0,0 +1,153 @@
+/**
+ * @flow
+ * @file Helper for the plain Box Upload API
+ * @author Box
+ */
+
+import noop from 'lodash/noop';
+import BaseUpload from './BaseUpload';
+import type { BoxItem } from '../../flowTypes';
+
+class PlainUpload extends BaseUpload {
+    successCallback: Function;
+    progressCallback: Function;
+
+    /**
+     * Handles an upload success response
+     *
+     * @param {Object} data - Upload success data
+     * @return {void}
+     */
+    uploadSuccessHandler = ({ entries }: { entries: BoxItem[] }): void => {
+        if (this.isDestroyed()) {
+            return;
+        }
+
+        if (typeof this.successCallback === 'function') {
```

Comment:
check needed when defaulted to noop?

Reply:
I just copy pasted PlainUpload.js into another directory, maybe @tonyjin can answer comments related to PlainUpload?